src/Microsoft.OData.Edm/Schema/EdmModelBase.cs

```diff
@@ -23,6 +24,11 @@ public abstract class EdmModelBase : EdmElement, IEdmModel
         private readonly Dictionary<string, IEdmTerm> termDictionary = new Dictionary<string, IEdmTerm>();
         private readonly Dictionary<string, IList<IEdmOperation>> functionDictionary = new Dictionary<string, IList<IEdmOperation>>();
 
+        /// <summary>
+        /// Cache of operations that are bindable to entity types, its a cache of all bindable functions, indexed by binding type.
+        /// </summary>
+        private readonly ConcurrentDictionary<string, IList<IEdmOperation>> bindableOperationsCache;

```

Comment:
With these new caches introduced, there shouldn't be a problem with it growing too big during process lifetime, where we need to start considering a eviction policy or something of that sort right?

Reply:
Shouldnt be the case as this cache will contain only a subset of operations

Reply:
nope; max size will be the number of bindable operations in the model; they are just partitioned by type.

---
In reply to: [493171544](https://github.com/OData/odata.net/pull/1888#discussion_r493171544) [](ancestors = 493171544)